acceptance/tests/language/pcore_generate_env_isolation.rb

```diff
@@ -0,0 +1,66 @@
+test_name 'C98345: ensure pcore generate assures env. isolation' do
+  require 'puppet/acceptance/environment_utils.rb'
+  extend Puppet::Acceptance::EnvironmentUtils
+
+  app_type        = File.basename(__FILE__, '.*')
+  tmp_environment = mk_tmp_environment_with_teardown(master, app_type)
+  tmp_environment2 = mk_tmp_environment_with_teardown(master, app_type)
+  fq_tmp_environmentpath  = "#{environmentpath}/#{tmp_environment}"
+  fq_tmp_environmentpath2  = "#{environmentpath}/#{tmp_environment2}"
+
+  type_name = 'conflicting'
+  relative_type_dir  = 'modules/conflict/lib/puppet/type'
+  relative_type_path = "#{relative_type_dir}/#{type_name}.rb"
+  on(master, "mkdir -p #{fq_tmp_environmentpath}/#{relative_type_dir}")
```

Comment:
+1. The "prep" work is a step, too, albeit a "pre-step". The `step` structure improves readability for me.